src/views/jobManager/selfCodes/selfCodesResultsView.ts

```diff
@@ -0,0 +1,155 @@
+
+export class selfCodesResultsView implements TreeDataProvider<any> {
+  private _onDidChangeTreeData: EventEmitter<ChangeTreeDataEventType> =
+    new EventEmitter<ChangeTreeDataEventType>();
+  readonly onDidChangeTreeData: vscode.Event<ChangeTreeDataEventType> = this._onDidChangeTreeData.event;
+  private autoRefresh: boolean = false;
+  constructor(context: vscode.ExtensionContext) {
+    context.subscriptions.push(
+      vscode.commands.registerCommand(`vscode-db2i.self.refresh`, async () => this.refresh()),
+      vscode.commands.registerCommand(`vscode-db2i.self.reset`, async () => {
+        const selected = JobManager.getRunningJobs();
+        if (selected) {
+          try {
+            const resetSelfCmd = `DELETE FROM qsys2.sql_error_log where user_name = current_user`
+            await JobManager.runSQL(resetSelfCmd, undefined);
+            this.refresh();
+            vscode.window.showInformationMessage(`Reset SELF code error log.`)
+          } catch (e) {
+            vscode.window.showErrorMessage(`An Error occured reseting SELF code error log: ${e}`)
+          }
+        }
+      }),
+      vscode.commands.registerCommand(`vscode-db2i.self.enableAutoRefresh`, async () => {
+        this.autoRefresh = true;
+        vscode.commands.executeCommand(`setContext`, `vscode-db2i.self.autoRefresh`, true);
+        vscode.window.showInformationMessage("SELF Code Auto Refresh Enabled");
+      }),
+      vscode.commands.registerCommand(`vscode-db2i.self.disableAutoRefresh`, async () => {
+        this.autoRefresh = false;
+        vscode.commands.executeCommand(`setContext`, `vscode-db2i.self.autoRefresh`, false);
+        vscode.window.showInformationMessage("SELF Code Auto Refresh Disabled");
+      })
+    );
+    setInterval(async () => {
+      if (this.autoRefresh) {
+        this.refresh();
+      }
+    }, 5000);
+
+  }
+
+  async getSelfCodes(): Promise<SelfCodeNode[]> {
+    const selected = JobManager.getRunningJobs();
+    if (selected) {
+      const content = `SELECT job_name, user_name, logged_time, logged_sqlstate, logged_sqlcode, matches, stmttext, 
+                          message_text, message_second_level_text 
+                      FROM qsys2.sql_error_log, lateral 
+                          (select * from TABLE(SYSTOOLS.SQLCODE_INFO(logged_sqlcode)))
+                      where user_name = current_user
+                      order by logged_time desc`;
+      const data: SelfCodeNode[] = await JobManager.runSQL<SelfCodeNode>(content, undefined);
+      return data;
+    }
+    return;
+  }
+
+  refresh(): void {
+    this._onDidChangeTreeData.fire();
+  }
+
+  getTreeItem(element: any): TreeItem | Thenable<TreeItem> {
+    return element;
+  }
+  async getChildren(element?: any): Promise<any[]> {
+    if (element) {
+      return [];
+    } else {
+      const selfCodes = await this.getSelfCodes();
+      return selfCodes.map((error) => {
```

Comment:
There is a chance that `selfCodes` can be undefined here, as `getSelfCodes` can return `undefined` if no job is running. Considering returning an empty array so then no items will show up, and then use `viewsWelcome` in the `package.json` file to have default text.

Reply:
I added the if statement to make sure it is defined!